diesel/src/serialize.rs

```diff
@@ -76,7 +76,7 @@ impl<DB: TypeMetadata> Output<'static, Vec<u8>, DB> {
     /// Unsafe to use for testing types which perform dynamic metadata lookup.
     pub fn test() -> Self {
         use std::mem;
-        #[allow(clippy::invalid_ref)]
+        #[allow(clippy::invalid_ref, deprecated)]
```

Comment:
Why do we allow this?

Reply:
The alternative for the line below would be `Self::new(Vec::new(), unsafe { mem::MaybeUninit::uninit().assume_init() })` which would basically be also UB.
Therefore I think it is not worth changing it. 
(cc @diesel-rs/core just in case)

Reply:
Ah okay, I got it. Could we add a comment about it?

Reply:
A comment for what exactly? There is already a comment describing what this function tries to achieve.

Reply:
I mean it's reasonable to leave a comment on why we use the deprecated method, what about?